1_fetch.R

```diff
@@ -41,7 +42,30 @@ p1_targets_list <- list(
     p1_nhd_reaches,
     download_nhdplus_flowlines(p1_drb_comids_all_tribs$COMID)
   ),
-
+  
+  ## May take a while to run
+  tar_target(
+    p1_nhd_catchments,
+    get_nhdplusv2_catchments(comid = p1_nhd_reaches$comid)
+  ),
+  
+  tar_target(
+    p1_nhd_catchments_along_nhm,
+    get_nhdplusv2_catchments(comid = p1_nhd_reaches_along_NHM$comid)
+  ),
+  
+  tar_target(
+    p1_nhm_catchments_upstream,
+    {sf_use_s2(FALSE)
+      left_join(p1_nhd_catchments %>% mutate(COMID = as.character(COMID)),
+                p1_drb_comids_all_tribs %>% mutate(COMID = as.character(COMID)),
+                by = 'COMID') %>%
+        group_by(segidnat) %>%
+        dplyr::summarize()
```

Comment:
I didn't realize a call to `summarize()` would work for this use case - good to know! I'd recommend making the summarize/union step more explicit though to increase the readability of this code chunk:
```suggestion
        dplyr::summarize(geometry = sf::st_union(geometry))
```